### DISCUSSION TITLE
Only owner will be able to push to master/approve PR

### DISCUSSION THREAD

**Discussion by AmosSzust** [community/community#35694] [Pull Requests]
Hi
I own a repository and I want to be to only one who can push to master or approve PRs.

I saw an option "Do not allow bypassing the above settings". When checked, I couldn't do both (can't approve my own PR). When unchecked, I could do both, but I don't know if other users will be able to do the same, which is not good.

I created a new file called CODEOWNERS as instructed. I didn't check it yet but since it's part of the repo, other people can edit it, no?

I want only **me** (the owner) to either push or approve PR, is it just a matter of a setting, or more than that (like CODEOWNERS, etc)?
Thanks
Amos

**Comment by fadyat** [comment]
Hello, set this in settings:

<img width="444" alt="Screenshot 2022-10-10 at 5 40 16 PM" src="https://user-images.githubusercontent.com/70910148/194891913-19c61571-ad55-4991-adb1-caeead9d0d16.png">

You can bypass, if you only have owner role, these rules if you do not check the box
```
Do not allow bypassing the above settings
The above settings will apply to administrators and custom roles with the "bypass branch protections" permission.
```
--- 
Create this rule and a CODEOWNERS. It's all.
If people will want to edit, they will need to create PRs and forks of your repo

**Comment by AmosSzust** [comment]
Hi @fadyat and thank for replying.

I just want to make sure I got it right: 
Now there's only an owner but later on **regular / contributors** users will join.
I already created CODEOWNERS file with `* @AmosSzust` 
So checking this option "Do not allow bypassing the above settings" will make sure only I will be able to approve PRs?

Thanks

**Reply by fadyat** [reply]
This ensures that only when you approve a PR, it can be merged.
<img width="333" alt="Screenshot 2022-10-10 at 6 13 14 PM" src="https://user-images.githubusercontent.com/70910148/194898963-c644df1b-b11e-418c-9c6c-60d6190cb4ef.png">

**Reply by fadyat** [reply]
More about [protected branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-pull-request-reviews-before-merging)

**Comment by AmosSzust** [comment]
I did as described including the CODEOWNERS file but I still can't commit to master nor approve my own pull requests. I checked this option "Do not allow bypassing the above settings" because I don't want anyone else to "bypass" the rules. 

In other words, how can I make sure that no one else is bypassing the CODEOWNERS rule, assuming the option "Do not allow bypassing the above settings" is **unchecked**?

**Comment by ossanna16** [comment]
Thank you for asking a great question @AmosSzust! I'm glad you found our community 🚀
